packages/rocketchat-lib/server/models/Rooms.js

```diff
@@ -767,6 +767,15 @@ class ModelRooms extends RocketChat.models._Base {
 
 		return this.remove(query);
 	}
+
+	removeUsernameById(roomId, username) {
```

Comment:
The `usernames` array is no longer used inside rooms, because of this the method has been removed from `Rooms` model. What you really need to do is to remove the `servedBy` field from the room, so I suggest you to create a new method on Livechat `Room` to remove the this field. You can find the Livechat `Room` model here: https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-livechat/server/models/Rooms.js
After creating the new method you need to replace the `removeUsernameById` by your new method.